src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -46,6 +50,49 @@ export class StacheNavService {
     return /^(https?|mailto|ftp):+|^(www)/.test(path);
   }
 
+  public updateRoutesOnScroll(routes: StacheNavLink[]) {
+    this.updateView(routes);
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: StacheNavLink[]) {
+    this.trackPageOffset();
+    this.isCurrent(routes);
+  }
+
+  // Checks to see if stache-wrapper exists and provides offset from top of window
+  private getHeaderHeight() {
+    const stacheWrapper = this.windowRef.nativeWindow.document.querySelector('.stache-wrapper');
+
+    if (stacheWrapper) {
+      this.headerHeight = stacheWrapper.offsetTop;
+    }
+  }
+
+  private trackPageOffset() {
+    this.getHeaderHeight();
+
+    // Represents top of page
+    // headerHeight buffer removes discrepency when hero or other 'header' elements exist
+    // 50px buffer aids in click-to-anchor and user visualization
```

Comment:
We can use the `StacheOmnibarAdapterService` to determine if the Omnibar exists.  This service also has the height of the omnibar `50` stored within it.  For components that need to change behavior or height based on the omnibars height, we want to use the value from that service.  This way, in the event the omnibar changes height, we can update the value in one place and all affected components will be updated.

Reply:
After looking back in to this logic the code is actually interpreting not only the omnibar height, but also the Stache Hero height, since the Stache Hero is meant to sit outside of the Stache Wrapper.

Reply:
I think the Stache Hero is accounted for with the `headerHeight`, the 50px is from the omnibar, and should use the StacheOmnibarAdapaterService to get that height value.